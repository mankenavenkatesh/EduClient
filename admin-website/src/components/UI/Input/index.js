import React from "react";
import PropTypes from "prop-types";
/** @jsx jsx */
import { css, jsx } from "@emotion/core";
import { invalidColor, black } from "../../../styles/variables";

const base = css`
  padding: 0.5rem 1rem;
  border: solid 1px black;
  outline: none;
  width: 100%;
`;

const rounded = css`
  border-radius: 5px;
`;

const pill = css`
  border-radius: 50px;
`;

const Input = ({
  custom,
  variant,
  type,
  onChange,
  value,
  message,
  ...rest
}) => (
  <React.Fragment>
    <input
      css={css(
        base,
        custom,
        variant === "rounded" && rounded,
        variant === "pill" && pill
      )}
      variant={variant}
      type={type}
      style={{ borderColor: message ? invalidColor : black }}
      onChange={onChange}
      value={value}
      size={50}
      {...rest}
    />
    {message && <small style={{ color: invalidColor }}>{message}</small>}
  </React.Fragment>
);

export default Input;

Input.propTypes = {
  message: PropTypes.string,
  custom: PropTypes.object,
  className: PropTypes.string,
  style: PropTypes.object,
  rest: PropTypes.object,
  onChange: PropTypes.func,
  value: PropTypes.string,
  variant: PropTypes.string,
  type: PropTypes.string
};
