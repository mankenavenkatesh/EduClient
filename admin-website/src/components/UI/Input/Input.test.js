// Input.spec.js
import React from "react";
import jest from "jest-mock";
import enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { matchers } from "jest-emotion";
import renderer from "react-test-renderer";
import Input from ".";

expect.extend(matchers);

describe("Input component", () => {
  it("should be defined", () => {
    expect(Input).toBeDefined();
  });

  it("shows Input text", () => {
    const input = renderer.create(<Input>Test Input</Input>);
    expect(input.toJSON()).toMatchSnapshot();
  });

  it("shows pill Input type", () => {
    const input = renderer.create(<Input variant="pill">Test Input</Input>);
    const tree = input.toJSON();
    expect(tree).toHaveStyleRule("border-radius", "50px");
  });

  it("shows rounded Input type", () => {
    const input = renderer.create(<Input variant="rounded">Test Input</Input>);
    const tree = input.toJSON();
    expect(tree).toHaveStyleRule("border-radius", "5px");
  });

  it("should call mock function when Input is changed", () => {
    enzyme.configure({ adapter: new Adapter() });
    const mockFn = jest.fn();
    const component = enzyme.mount(<Input onChange={mockFn} />);
    component.find("input").simulate("change");
    expect(mockFn).toBeCalled;
  });
});
