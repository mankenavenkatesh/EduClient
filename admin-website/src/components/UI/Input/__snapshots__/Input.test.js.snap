// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component shows Input text 1`] = `
<input
  className="css-d36eou-Input"
  size={50}
  style={
    Object {
      "borderColor": "#000000",
    }
  }
>
  Test Input
</input>
`;
